src/runtime_src/core/edge/common_em/config.cxx

```diff
@@ -795,4 +843,79 @@ namespace xclemulation{
     ifs.close();
     return false;
   }
+
+  std::string getXclbinVersion(const axlf* top)
+  {
+    std::string sVersion = "";
+    const axlf_section_header *xml_hdr = ::xclbin::get_axlf_section(top, BUILD_METADATA);
+
+    if (!xml_hdr) {
+      return sVersion;
+    }
+    auto begin = reinterpret_cast<const char*>(top) + xml_hdr->m_sectionOffset;
+    const char *json_data = reinterpret_cast<const char*>(begin);
+    uint64_t xml_size = xml_hdr->m_sectionSize;
+
+    boost::property_tree::ptree json_project;
+    std::stringstream json_stream;
+    json_stream.write(json_data, xml_size);
+    boost::property_tree::read_json(json_stream, json_project);
+    auto json_meta = json_project.get_child_optional("build_metadata");
+    if (!json_meta) {
+      return sVersion;
+    }
+    auto buildMetaData = json_project.get_child("build_metadata");
+    //std::string sTool = buildMetaData.get<std::string>("xclbin.generated_by.name", "");
+    sVersion = buildMetaData.get<std::string>("xclbin.generated_by.version", "");
+    //std::string sTimeStamp = buildMetaData.get<std::string>("xclbin.generated_by.time_stamp", "");    
+    //std::cout << __func__ <<" Tool : " << sTool << " Version : " << sVersion << " TimeStamp : " << sTimeStamp << std::endl;
+    return sVersion;
+  }
+
+  std::string getVivadoVersion()
+  {
+    char *xilinxVivadoEnvvar = getenv("XILINX_VIVADO");
+    std::string sVivadoDir = "";
+    if (xilinxVivadoEnvvar)
+    {
+      sVivadoDir = xilinxVivadoEnvvar;
+    }
```

Comment:
It seems the use of '{ ... }' in this file is inconsistent.   Recommended practice in XRT is no curly-brace if single statement, followed by empty new line.  Or curly brace immediately on same line as condition.

Short circuit this function if sVivadoDir.emty() on line 883

I am starring at this function and have no idea what it is doing?  It loops [2000,2100] * 4 times to compute a version string, but I am not sure what it is doing.  Seems way too complicated and lacks comments.

Reply:
This function not needed. So, removed from the code.